.github/workflows/build.yaml

```diff
@@ -66,13 +65,55 @@ jobs:
         with:
           path: ./website/public
 
+  api-docs:
+    uses: ./.github/workflows/api-docs.yaml
+    secrets: inherit
+
+  adr-index:
+    uses: ./.github/workflows/adr-index.yaml
+    secrets: inherit
+
+  # Build job
+  architecture:
+    needs:
+      - api-docs
+      - adr-index
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          path: tmp/architecture
+          repository: konflux-ci/architecture
+      - name: Download generated API docs
+        uses: actions/download-artifact@v3
+        with:
+          name: api-docs
+          path: tmp/architecture/ref/
+      - name: Download generated ADR index
+        uses: actions/download-artifact@v3
+        with:
+          name: adr-index
+          path: tmp/architecture/ADR/
+      - name: Setup Pages
+        uses: actions/configure-pages@v2
+      - name: Build with Jekyll
+        uses: actions/jekyll-build-pages@v1
+        with:
+          source: ./tmp/architecture/
+          destination: ./_site
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v1
```

Comment:
We're using v3 of this action for hugo. I think we should have align this one with that.